include/oneapi/tbb/detail/_concurrent_unordered_base.h

```diff
@@ -727,13 +727,13 @@ class concurrent_unordered_base {
         iterator end() const { return iterator(my_instance.first_value_node(my_end_node)); }
 
         const_range_type( const concurrent_unordered_base& table )
-            : my_instance(table), my_begin_node(const_cast<node_ptr>(&table.my_head)), my_end_node(nullptr)
+            : my_instance(table), my_begin_node(my_instance.first_value_node(const_cast<node_ptr>(&table.my_head))), my_end_node(nullptr)
```

Comment:
Why does the my_begin_node should be initialized with pointer to value_node? Is it some sort of optimization?

Reply:
My understanding was to fit the logic in `empty`, but now I see that the logic in `empty` can be reverted, because `my_begin_node` cannot be dummy.